packages/flutter/lib/src/material/time_picker_theme.dart

```diff
@@ -102,7 +103,21 @@ class TimePickerThemeData with Diagnosticable {
   /// brightness is [Brightness.dark].
   /// If the segment is not selected, [Colors.transparent] is used to allow the
   /// [Dialog]'s color to be used.
-  final Color? dayPeriodColor;
+  Color? get dayPeriodColor {
+    if (_dayPeriodColor == null || _dayPeriodColor is MaterialStateColor) {
+      return _dayPeriodColor;
+    }
+    return MaterialStateColor.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.selected)) {
+        return _dayPeriodColor;
+      }
+      // The unselected day period should match the overall picker dialog color.
+      // Making it transparent enables that without being redundant and allows
+      // the optional elevation overlay for dark mode to be visible.
+      return Colors.transparent;
```

Comment:
nit: Should we here just provide `return null;` so the unselected `dayPeriodColor` will be the default `dayPeriodColor`? They should have same effect(both transparent), but just in case MaterialDesign tokens are changed in the future:)

Reply:
Unfortunately, `resolveWith` requires a non-null return value, so I can't return null here.

Reply:
Ah I see. That makes sense:) Thanks!